enable using mouse on the radial menu

rafccq opened the pull request:
Solves #107

Commit:
enable using mouse on the radial menu

fgsfdsfgs commented:
What's the whole current device thing for? Is it not enough to just check if `dx`/`dy` are non-zero, like the rest of the input code does?

NeonNyan commented:
I'd really like to be able to play with a left-analog stick and mouse for aiming. This would explicitly break that.

When the check is removed, you can't move the radial dial with the xbox sticks after using the mouse unless you close and re-open the radial menu.

NeonNyan commented:
> I'd really like to be able to play with a left-analog stick and mouse for aiming. This would explicitly break that.
> 
> When the check is removed, you can't move the radial dial with the xbox sticks after using the mouse unless you close and re-open the radial menu.

Also it's not just this one case: it's preferable to map the steam deck gyro / stick / trackpads to the mouse, which would cause unfriendly behavior unless key presses are being handled correctly in the game code.

fgsfdsfgs commented:
That can be solved by just adding up movement from the sticks and the mouse instead of making them exclusive. Let me see what I can do.

rafccq commented:
@fgsfdsfgs I didn't use dx/dy to keep the slot selected, and not return to the center if there was no movement.
I added the device thing so the game detects the last device you used, and use only that one for the radial menu input, but
yes sounds better to just add the movements of mouse and sticks, I didn't think of it. 
That will make it work with control schemes such as the ones mentioned by @NeonNyan

rafccq commented:
Thanks, I'll test it out and tune if needed.

Commit:
Merge branch 'base-port' into port

Commit:
enable using mouse on the radial menu: reworked

mouse+kb and other input devices are no longer exclusive when using the radial menu

Commit:
enable using mouse on the radial menu: removed unused constants

Commit:
attempt to fix whitespace issues

AXDOOMER commented:
The new code doesn't use the same indentation as the rest of the file. This should be fixed.

AXDOOMER commented on port/src/input.c:
```diff
@@ -479,7 +480,7 @@ static inline s32 inputAxisScale(s32 x, const s32 deadzone, const f32 scale)
 		return 0;
 	} else {
-		// rescale to fit the non-deadzone range
+	 	// rescale to fit the non-deadzone range
```
This new random whitespace should be removed.

AXDOOMER commented on src/game/activemenutick.c:
```diff
@@ -63,6 +67,24 @@ void amTick(void)
+                    cstickx = (newstickx < -128) ? -128 : (newstickx > 127) ? 127 : newstickx;
+                    csticky = (newsticky < -128) ? -128 : (newsticky > 127) ? 127 : newsticky;
+                }
+#endif
```
The new code doesn't use the same indentation as the rest of the file. This should be fixed.

AXDOOMER commented on src/game/activemenutick.c:
```diff
@@ -337,6 +359,12 @@ void amTick(void)
+            g_AmMenus[g_AmIndex].mousex = 0.f;
+            g_AmMenus[g_AmIndex].mousey = 0.f;
+        }
+#endif
```
This code is also not indented the same as the rest of the code. This file uses tabs put your editor is inserting spaces.

AXDOOMER commented on src/include/types.h:
```diff
@@ -4306,6 +4306,11 @@ struct activemenu {
+#ifndef PLATFORM_N64
+    /*    */ f32 mousex;
+    /*    */ f32 mousey;
+#endif
```
Also not indented as the rest.

fgsfdsfgs commented:
The whitespace issues are because you're using spaces instead of tabs, unlike the rest of the code.

Commit:
fix whitespace issues

Commit:
Merge remote-tracking branch 'base/port' into port

Commit:
Merge remote-tracking branch 'base/port' into port
